### ISSUE TITLE
Servo build fails in build farm (Rolling)

### ISSUE THREAD

**Issue by henningkayser** [ros-planning/moveit2#1826]
Servo fails to compile the Rolling binaries (build farm) due to a missing `ros2_control` header https://build.ros2.org/job/Rbin_uJ64__moveit_servo__ubuntu_jammy_amd64__binary/89/console

```
08:22:20 /tmp/binarydeb/ros-rolling-moveit-servo-2.6.0/src/servo_calcs.cpp:45:10: fatal error: controller_manager/realtime.hpp: No such file or directory
08:22:21    45 | #include <controller_manager/realtime.hpp>
08:22:21       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
08:22:21 compilation terminated.
08:22:21 make[4]: *** [CMakeFiles/moveit_servo_lib.dir/build.make:121: CMakeFiles/moveit_servo_lib.dir/src/servo_calcs.cpp.o] Error 1
08:22:21 make[4]: Leaving directory '/tmp/binarydeb/ros-rolling-moveit-servo-2.6.0/.obj-x86_64-linux-gnu'
08:22:21 make[3]: *** [CMakeFiles/Makefile2:206: CMakeFiles/moveit_servo_lib.dir/all] Error 2
08:22:21 make[3]: Leaving directory '/tmp/binarydeb/ros-rolling-moveit-servo-2.6.0/.obj-x86_64-linux-gnu'
08:22:21 make[2]: *** [Makefile:149: all] Error 2
08:22:21 make[2]: Leaving directory '/tmp/binarydeb/ros-rolling-moveit-servo-2.6.0/.obj-x86_64-linux-gnu'
08:22:21 dh_auto_build: error: cd .obj-x86_64-linux-gnu && make -j1 VERBOSE=1 returned exit code 2
08:22:21 make[1]: *** [debian/rules:44: override_dh_auto_build] Error 25
08:22:21 make[1]: Leaving directory '/tmp/binarydeb/ros-rolling-moveit-servo-2.6.0'
08:22:21 make: *** [debian/rules:27: build] Error 2
08:22:21 dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
08:22:21 E: Building failed
```

The header has been removed with https://github.com/ros-controls/ros2_control/pull/794, and then the breaking change has been released into Rolling.

The fix is obviously to remove ros2_control from our `.repos` dependencies and to fix whatever has to be fixed with the latest changes. The real issue however is that we shouldn't keep fixed third-party dependencies in our `.repos` file at all. It's somewhat ok for repositories where we synchronize the release process (everything in the MoveIt world that we maintain). Everything else should be build against the debians so that we can detect and fix breaking upstream changes in time when we hit a breaking testing build in CI, and not when we hit a breaking release build in the build farm.

**But how do we add support for new features in third-party dependencies like ros2_control?**
There are two options:
* The new feature has to be implemented in a way that is backwards compatible in MoveIt, so when `ros2_control` bumps the release we are already set. There are plenty of ways to do this, use include guards, version-based compile directives etc...
* The breaking change has to sit in a feature branch or fork until the upstream package is released

In short, we want to fail release builds in our testing CI job, because that's the right time to fix things for the next release. If we have the need to keep track with ongoing development in other packages, we can talk about adding the package to repos and add a staging build that keeps track of the latest head (not a fixed release tag that gets outdated the next month).

**Comment by AndyZe** [comment]
Note that our instructions call for building from source:

https://moveit.ros.org/install-moveit2/source/

Your proposed policy sounds OK but the source build is often going to fail, unfortunately.

**Comment by AndyZe** [comment]
Really the issue comes down to, do we prioritize debians or do we prioritize building from source?

**Comment by henningkayser** [comment]
I'm not against building from source, but we should build the latest versions from source and not depend on outdated versions in our workspace.
The priority should always be functioning debians in the end. If we detect breakages on the build farm, it's way too late to fix it without stress. By removing third-party repos from our workspace, we can make sure that we will always be noticed about upstream breakages in CI (testing). If we still think we won't have enough time, I'm proposing an extra CI job that tests against not-yet-released versions (=head). This policy would make our source builds fail early (in CI) so that we can fix them instead of having to fix releases later.

**Comment by henningkayser** [comment]
continued in #1840
